code/game/machinery/dna_scanner.dm

```diff
@@ -132,13 +135,53 @@
 	if(default_deconstruction_crowbar(I))
 		return
 
+	if(panel_open && is_wire_tool(I))
+		wires.interact(user)
+		return
+
 	return ..()
 
 /obj/machinery/dna_scannernew/interact(mob/user)
 	toggle_open(user)
 
 /obj/machinery/dna_scannernew/MouseDrop_T(mob/target, mob/user)
 	var/mob/living/L = user
-	if(user.stat || (isliving(user) && (!(L.mobility_flags & MOBILITY_STAND) || !(L.mobility_flags & MOBILITY_UI))) || !Adjacent(user) || !user.Adjacent(target) || !iscarbon(target) || !user.IsAdvancedToolUser())
+	if(user.stat || (isliving(user) && (!(L.mobility_flags & MOBILITY_STAND) || !(L.mobility_flags & MOBILITY_UI))) || !Adjacent(user) || !user.Adjacent(target) || !iscarbon(target) || !user.IsAdvancedToolUser() || locked)
 		return
 	close_machine(target)
+
+/obj/machinery/dna_scannernew/proc/irradiate(mob/living/carbon/target)
+	if(!HAS_TRAIT(target, TRAIT_RADIMMUNE))
+		to_chat(target, "<span class='danger'>You feel warm.</span>")
+		target.rad_act(250/(damage_coeff ** 2))
+
+		if(target.has_dna() && !HAS_TRAIT(target, TRAIT_BADDNA))
+			var/resist = target.getarmor(null, "rad")
+			if(prob(max(0,100-resist)))
+				target.randmuti()
+				if(prob(20))
+					if(prob(90))
```

Comment:
🤨

Reply:
This just means that there's a 20% chance of getting a mutation and 90% of the time it's negative